extmod/network_cyw43.c

```diff
@@ -280,13 +292,19 @@ STATIC mp_obj_t network_cyw43_connect(size_t n_args, const mp_obj_t *pos_args, m
     if (ret != 0) {
         mp_raise_OSError(-ret);
     }
+    #ifdef hal_set_network_active
+    hal_set_network_active(MP_QSTR_WLAN, true);
+    #endif
     return mp_const_none;
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_KW(network_cyw43_connect_obj, 1, network_cyw43_connect);
 
 STATIC mp_obj_t network_cyw43_disconnect(mp_obj_t self_in) {
     network_cyw43_obj_t *self = MP_OBJ_TO_PTR(self_in);
     cyw43_wifi_leave(self->cyw, self->itf);
+    #ifdef hal_set_network_active
+    hal_set_network_active(MP_QSTR_WLAN, false);
+    #endif
```

Comment:
As above, shouldn't be needed, because you call `.active(False)` to turn of the WiFi interface.

Reply:
See other comments.